javascript/javascript1/week3/smartPhoneApp.js

```diff
@@ -0,0 +1,141 @@
+//             if(usageOfToday >= 120) {
+//                 return 'You have reached your limit, no more smartphoning for you!'
+//             }else{
+//                 return `You have added ${activities.length} activities. They amount to ${timeOfUsage} min. of usage.`
+//             }
+//         }else{
+//             let timeOfUsage = 0;
+//             for(i = 0; i < activities.length; i++){
+//                 timeOfUsage += activities[i].duration;
+//             }
+//             return `You have added ${activities.length} activities. They amount to ${timeOfUsage} min. of usage.`
+//         }
+//     }
+// }
+
+// console.log(showStatus());
+// // addActivity('24/11-26', 'Do Homework', 100);
+// // addActivity('24/11-26', 'Coding', 100);
+// console.log((activities));
+
+
+
+//  input Date automatically
+const activities = [];
+
+// I tried to as order of date, content,then doration, it led content and duration move forward, so here I put date in the end.
+function addActivity(activityContent, activityDurationMin, activityDateInput) { 
+    const activityDate = activityDateInput || new Date().toLocaleDateString("en-US"); // input a specify day or automatically today's date
+    let activityInfo = {
+        date: new Date(activityDate).toLocaleDateString("en-US"),
+        activity: activityContent,
+        duration: activityDurationMin,
+    };
+    activities.push(activityInfo);
+}
+addActivity("Youtube", 130, "11/26/2024");
+addActivity("Youtube", 30);
+addActivity("Facebook", 20);
+addActivity("Homework", 150)
+console.log(activities)
+
+// as date format is already unified, then showStatus should be updated. Here I would like to show status of today or a specific day.
+// here I want to alert only for the usage limit of today, if the previous records sum over limit, it won't alert but a summary of usage.
+function showStatus(dateInput) {
+    if(activities.length === 0){
+        return 'Add some activities before calling showStatus.'
+    }else{
+        let activitiesOneDay = activities.filter(activities => activities.date === dateInput);
+        if(activitiesOneDay.length > 0){
+            let usageOfOneDay = activitiesOneDay.reduce((totalDuration, activities) => totalDuration + activities.duration, 0);
+            if(usageOfOneDay >= 120 && dateInput === new Date().toLocaleDateString("en-US")) {
+                return 'You have reached your limit, no more smartphoning for you!'
+            }else{
+                return `You have added ${activitiesOneDay.length} activities today. They amount to ${usageOfOneDay} min. of usage.`
+            }
+        }else{
+            return `Add some activities on ${dateInput} before calling showStatus.`
+        }
+    }
+}
+
+const checkToday = new Date().toLocaleDateString("en-US");
+console.log(showStatus(checkToday)); //alert when sum of today are over limit.
```

Comment:
Excellent tests! 👍 I hope dates haven't scared you away; they can be terrible to work with. 🥴

Reply:
Yes!!! Almost two weeks, I was struggling with dates! This idea actually came from the last online session, when Thomas explained to me. Thank you both!